File: info/1. Алгоритмы/5. Вычислительные алгоритмы/6. Слияние отсортированных последовательностей/code sample/Sample1.java
package sample;

import java.util.Arrays;

public class Sample1 {

	public static void main(String[] args) {

		int[] firstArray = new int[] { 1, 3, 6, 9 };
		int[] secondArray = new int[] { 2, 4, 4, 7 };

		int[] result = merge(firstArray, secondArray);

		System.out.println(Arrays.toString(result));

	}

	public static int[] merge(int[] firstArray, int[] secondArray) {
		int[] result = new int[firstArray.length + secondArray.length];
		int l = 0;
		int r = 0;
		for (int i = 0; i < result.length; i++) {
			if (l >= firstArray.length) {
				result[i] = secondArray[r];
				r += 1;
			} else if (r >= secondArray.length) {
				result[i] = firstArray[l];
				l += 1;
			} else if (firstArray[l] < secondArray[r]) {
				result[i] = firstArray[l];
				l += 1;
			} else {
				result[i] = secondArray[r];
				r += 1;
			}
		}
		return result;
	}

}

File: info/1. Алгоритмы/5. Вычислительные алгоритмы/6. Слияние отсортированных последовательностей/code sample/sample1.py
def merge(first_sequince, second_sequince):
    result = [0 for x in range(len(first_sequince)+len(second_sequince))]
    l = 0
    r = 0
    for i in range(len(result)):
        if l >= len(first_sequince):
            result[i] = second_sequince[r]
            r += 1
        elif r >= len(second_sequince):
            result[i] = first_sequince[l]
            l += 1
        elif first_sequince[l] < second_sequince[r]:
            result[i] = first_sequince[l]
            l += 1
        else:
            result[i] = second_sequince[r]
            r += 1
    return result


first_sequince = [1, 3, 6, 9]
second_sequince = [2, 4, 4, 7]

result = merge(first_sequince, second_sequince)

print(result)
